Analytics/Integrations/Kahuna/SEGKahunaIntegration.m

```diff
@@ -19,16 +60,40 @@ - (id)init {
     self.name = @"Kahuna";
     self.valid = NO;
     self.initialized = NO;
+    
+    _kahunaCredentialsKeys = [NSSet setWithObjects:       KAHUNA_CREDENTIAL_USERNAME,
+                                                          KAHUNA_CREDENTIAL_EMAIL,
+                                                          KAHUNA_CREDENTIAL_FACEBOOK,
+                                                          KAHUNA_CREDENTIAL_TWITTER,
+                                                          KAHUNA_CREDENTIAL_LINKEDIN,
+                                                          KAHUNA_CREDENTIAL_USER_ID,
+                                                          KAHUNA_CREDENTIAL_GOOGLE_PLUS,
+                                                          KAHUNA_CREDENTIAL_INSTALL_TOKEN, nil];
   }
   return self;
 }
 
 - (void)start {
-  [KahunaAnalytics launchWithKey:[self.settings objectForKey:@"apiKey"]];
-  [super start];
+  @try {
+    // We just need one call to launchWithKey and not multiple.
+    if ([KahunaPushMonitor sharedInstance].kahunaInitialized == NO) {
+      [KahunaAnalytics launchWithKey:[self.settings objectForKey:@"apiKey"]];
+      // If we have recorded any push user info, then
+      if ([KahunaPushMonitor sharedInstance].pushInfo != nil) {
+        [KahunaAnalytics handleNotification:[KahunaPushMonitor sharedInstance].pushInfo withApplicationState:[KahunaPushMonitor sharedInstance].applicationState];
+        [KahunaPushMonitor sharedInstance].pushInfo = nil;
+      }
+      
+      [KahunaPushMonitor sharedInstance].kahunaInitialized = TRUE;
+    }
+    
+    [super start];
+  }
+  @catch (NSException *exception) {
```

Comment:
Added for defensive purposes. Possibly an over-kill in this situation since we do check for the key to be a non-null string in the SDK anyway. Will remove the try-catch.